### ISSUE TITLE
React Content Query Webpart doesn't query the sites under the site collection

### ISSUE THREAD

**Issue by gurleen23** [SharePoint/sp-dev-fx-webparts#288]
#### Category
- [x] Question
- [ ] Bug
- [ ] Enhancement

#### Expected or Desired Behavior
As per the description of the webpart given [here](https://github.com/SharePoint/sp-dev-fx-webparts/tree/master/samples/react-content-query-webpart), the WebPart uses the search in order to get all sites under the current domain, which makes it possible to query not only subsites but other site collections and their subsites as well.

#### Observed Behavior
But, what I see after the webpart is deployed in app-catalog of the tenant I'm working on is something like this:

![image](https://user-images.githubusercontent.com/19613374/29461847-38e1236e-83fb-11e7-9d0f-db4c6cf4d10c.png)

I don't see the any subsites or any site collection here.
Can anyone help me with this please?

Thank You!

**Comment by schneika** [comment]
I am not able to retrieve single Teamsites (Groups)
An error occured while loading fields : The previously configured web url '/sites/webname' is not found anymore. Either leave the WebPart properties as is or select another web url.

**Comment by spplante** [comment]
Hi gurleen23, 

These relative URLs that you showed here should all be site collections under the current domain name. Let's say you added the WebPart on site with the url **https://mysite.sharepoint.com**, then then **/news** path shown in the dropdown would correspond to **https://mysite.sharepoint.com/news** which should normally be a site collection.

As for the sub sites, are you trying to connect the WebPart to a sub site that has recently been added? Because since the dropdown is driven by the search, you need to wait for the search to crawl the newly added content and update its index. This can take quite some time in SPO.

Also make sure you have access to the said subsite, as the search is also security-trimmed.

Let me know if that helps

**Comment by spplante** [comment]
Hi @gurleen23,

I found a minor mistake that turned out to have a pretty big impact on the WebPart : The search query responsible for filling the **Web Url** dropdown within the WebPart's toolpane had no **rowLimit** parameter specified, causing the query to return a maximum of **10** results since **10** is the default **rowLimit** value when not specified.

I updated the code by adding a **rowLimit** of 500 results in order to really get all sites on which the current user has access. This should work much better now :)

The pull request #296 has been submitted, you can either access the updated code or .sppkg directly from [my fork](https://github.com/spplante/sp-dev-fx-webparts/tree/dev/samples/react-content-query-webpart), or wait for the pull request #296 to be merged in here to the master branch.

@VesaJuvonen You can close this issue as soon as #296 is merged :)

**Comment by VesaJuvonen** [comment]
Closing since #296 has been merged. Thx.

**Comment by spplante** [comment]
FYI @gurleen23, I just submitted my pull request #299 which now allows the user to select a site collection before selecting the actual subsite. This narrows down by a lot the amount of results that are displayed in each dropdown. 

I also modified the search request used for filling both properties by using a recursive promise approach, which basically loads urls in batch of 500 results (maximum allowed in one call by the REST API) until everything is loaded.

This removes the 500 items rowLimit bottleneck and allows the WebPart to search for unlimited results and avoid any kind of missing sites behavior :)
